test/functional/name_deterministic_salt.py

```diff
@@ -0,0 +1,57 @@
+#!/usr/bin/env python3
+# Licensed under CC0 (Public domain)
+
+# Test that name_new uses deterministic salts.
+
+from test_framework.names import NameTestFramework
+from test_framework.util import *
+
+class NameDeterministicSaltTest(NameTestFramework):
+
+    def set_test_params(self):
+        self.setup_clean_chain = True
+        self.setup_name_test ([[]])
+
+    def run_test(self):
+        node = self.nodes[0]
+        node.sethdseed(seed="cQDxbmQfwRV3vP1mdnVHq37nJekHLsuD3wdSQseBRA2ct4MFk5Pq")
+        assert(node.listwallets() != [])
+
+        node.generate(200)
+
+        self.log.info("Begin registration of name.")
+        new_name = node.name_new('d/wikileaks')
+
+        self.log.info("Make sure salt matches.")
+        assert_equal(new_name[1], '40d0f82281da919314fbb3331d4c43f5005ecc0e')
+
+        node.generate(12)
+        self.log.info("Leave both fields blank (null).")
+        node.name_firstupdate ("d/wikileaks")
+        node.generate(1)
+        self.log.info("Name registered; no value provided.")
+
+        self.log.info("Try without giving any args")
+        new_name = node.name_new('d/name2')
+        node.generate(12)
+        node.name_firstupdate ("d/name2")
+        node.generate(1)
```

Comment:
For `d/wikileaks`, we make sure it's a specific salt, as a regression test. For `d/nameX`, we just check it's consistent enough to be able to register.

`d/wikileaks` is used in the unit tests too as well as in Electrum, so that's the root cause. But it would be possible to change, yes.

Reply:
What I mean is, why is the second block needed at all?  What kind of bug would the second block uncover that the first does not?